data/transactions/logic/eval.go

```diff
@@ -932,13 +933,31 @@ func parseStackTypes(spec string) StackTypes {
 	if spec == "" {
 		return nil
 	}
-	types := make(StackTypes, len(spec))
-	for i, letter := range spec {
+	types := make(StackTypes, 0, len(spec))
+	for i := 0; i < len(spec); i++ {
```

Comment:
Note that this does not have the exact same behavior as `for i, letter := range spec {` from the old code.

I suppose it shouldn't break anything to iterate over bytes instead of runes, but it looks like this code was set up to use runes everywhere, so it seems only right to keep that consistency.

To keep the old behavior, I might suggest converting `spec` to `[]rune` like this answer indicates: https://stackoverflow.com/a/46117090

Reply:
If `spec` is `[]rune` is makes it more annoying to include the bad spec in a `panic`. And, since I'm modifying `i` in the loop, I still wouldn't be able to use the `range` operator.

You're right about consistency. I will change the type of `AllStackTypes` to `map[byte]StackType` and skip the conversion of `letter` to `rune`. That will guard against someone losing their mind and trying to use an emoji as character in `proto()`.

So now it is clear we are processing the string as a sequence of bytes, and runes do not appear at all.

I'll leave resolving to you if that works, else let me know what you think.